src/Chapter1.hs

```diff
@@ -636,7 +653,9 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit n 
+    | abs n < 10 = abs n
+    | otherwise = firstDigit (div (abs n) 10)
```

Comment:
Ideally, you can write this function a bit cleaner by calling `abs` only once. Notice, that you don't need to compare with absolute value each time. You can create a helper recursive function and use `abs` only once.